python/ray/tests/test_client_library_integration.py

```diff
@@ -0,0 +1,21 @@
+import sys
+
+import pytest
+
+import ray
+import ray.util.client.server.server as ray_client_server
+
+from ray.rllib.examples import rock_paper_scissors_multiagent
+
+
+def test_rllib_integration(call_ray_stop_only):
+    ray.init(num_cpus=1)
+    ray_client_server.serve("localhost:50051")
+    # Connection timeout. Help.
+    ray.util.connect("localhost:50051")
+
```

Comment:
Specifically, the relevant gRPC channel does not become ready in time
https://github.com/ray-project/ray/blob/b125cd95b45f2f53c15993a513208761bd0a4072/python/ray/util/client/worker.py#L110

Another piece of info is that using the linked context manager does work 
https://github.com/ray-project/ray/blob/b125cd95b45f2f53c15993a513208761bd0a4072/python/ray/util/client/ray_client_helpers.py#L9
i.e. this works
```
with ray_start_client_server() as client:
    pass
```
which is weird because it does basically the same thing: start server, connect client:
https://github.com/ray-project/ray/blob/b125cd95b45f2f53c15993a513208761bd0a4072/python/ray/util/client/ray_client_helpers.py#L18-L19
(I don't know how I would use that context manager here)

Reviewers, any suggestions on how to fix this or how else we could establish a Ray client connection here?

Reply:
why not just use `with ray_start_client_server() as client:` then?

Reply:
or `with ray_start_client_server() as ray:` 
, I guess.

Because I'm not sure how to get this `ray` to be the `ray` used in the actual rock paper scissors main().

Reply:
If I do 
```ray_client_server.serve("localhost:50051"); ray.util.connect("localhost:50051")```
in a Python shell, it works.
If I do it in a script, I get a connection timeout.

Reply:
can you try adding a sleep in between?

Reply:
I've tried it -- doesn't change the outcome.

Reply:
Could you check if you see the same behavior?

Reply:
Yeah, I see the same behavior.

Reply:
> (I don't know how I would use that context manager here)


I think you could do
```
del ray 
with ray_start_client_server() as ray:
    from ray.rllib.examples import rock_paper_scissors_multiagent
    rock_paper_scissors_multiagent.main()
```

Reply:
Not sure if the ray imported from within `main()` would be the right one.

It works out with a `subprocess` call to `ray start --head` (which starts the server) and then ray.util.connect.
Going to write a fixture that does this and tears things down.

Reply:
You don't need to use a specific ray after connecting, the normal `ray` will work.

Reply:
Ah, that makes sense -- got confused about the Python import system.
If I do, say
```
with ray_start_client_server() as _:
    rock_paper_scissors_multiagent.main()
```
then `_` and the objects doing the ray api calls inside of main are identical to the global api stub 
```
https://github.com/ray-project/ray/blob/a9ac4ad890fab4355f1e5db6998c6a3395fca08f/python/ray/util/client/__init__.py#L158
```